go/vt/mysqlctl/capabilityset.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2019 The Vitess Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ Detect server flavors and capabilities
+*/
+
+package mysqlctl
+
+type mysqlFlavor string
+
+const (
+	flavorMySQL   mysqlFlavor = "mysql"
+	flavorPercona mysqlFlavor = "percona"
+	flavorMariaDB mysqlFlavor = "mariadb"
+)
+
+// Mysqld is the object that represents a mysqld daemon running on this server.
+type CapabilitySet struct {
+	mysqld *Mysqld
```

Comment:
This makes a lot of sense. In my original plan it could have relied on other functionality (detect if mysqld_safe existed and thus if systemd should be used). But I've removed this from capabilities, since it is not related to `mysqld` versions: only packaging choices.